### DISCUSSION TITLE
Add in title assigned categories on post page

### DISCUSSION THREAD

**Discussion by oim37** [wintercms/wn-blog-plugin#49] [Q&A] [answered]
Hello,

I want to show on the article page, next to the title show which subcategory it belongs to, having previously created them:

news (Main category)
-company news (subcategory)
-articles (subcategory)

```
title = "Base article"
url = "/support/kb/:slug"
layout = "second"
is_hidden = 0

[blogPost]
slug = "{{ :slug }}"
categoryPage = "kb"
==
<section class="section">
	<div class="container">
	
		<div class="section_header">
		<div class="item-news-block__date">{{post.published_at|date('d.m.Y')}}</div>
			<h2 class="section_title"><b>{{ post.title }}</b></h2><br/>
			<div class="section_text">
			
			</div>
		</div>
		<div class="post">
		{{ post.content_html|raw }}<br/>
		</div>
		<a href="{{ 'novosti' | page }}" class="item-news-block__more link-line">
							{{ 'nav.all'|_ }}
		</a>
	</div>
</section>

```
I try this code near `{{ post.title }}` like  

`{{ this.category.name }}` or `{{ category.name }}` and nothing to show.

I check this code 

```
{% if not category %}
    <h2>Category not found</h2>
{% else %}
    <h2>{{ category.name }}</h2>

    {% component 'blogPosts' %}
{% endif %}
```

and it show Category not found. This article have two categories - news and articles.

And if it possible to show only assigned subcategories.

**Comment by mjauvin** [comment, marked as answer]
You need to use this:
```twig
<ul>
{% for category in post.categories %}
    <li>{{ category.name }}</li>
{% endfor %}
</ul>
```

**Reply by oim37** [reply]
Yes it works, thank you very much.

The fact is that news in this category falls through the news filter, i.e. first, is it possible to show only the subcategory of membership?

**Reply by mjauvin** [reply]
Not sure what you mean exactly

**Reply by oim37** [reply]
So it shows the presence of an article in 2 categories

News and Articles. The second (Articles) is a subcategory. Show only availability in subcategories, ignoring the news category.

**Reply by oim37** [reply]
The blog has a hierarchy setting, whether it is possible to ignore the display of the main category and display only the subcategory.

![image](https://github.com/wintercms/wn-blog-plugin/assets/6842168/510fa9f3-f171-4890-8f92-8018f3fedbd1)

**Comment by mjauvin** [comment]
Can't you just assign the subcategory and not its parent as the article categories ?

**Comment by oim37** [comment]
This is an option, but I then need to display 2 types of subcategories in one section.

Now it looks like:

```
[blogPosts]
categoryFilter = "news"
```

How make 2 categories in this? I try 

```
[blogPosts]
categoryFilter = "articles;corp-news"

```
and 

```
[blogPosts]
categoryFilter = "articles,corp-news"
```

It not working and show all in blog.

**Comment by mjauvin** [comment]
Oh, I see what you mean now. I guess you need to add more code in twig to only show leaves categories

**Comment by mjauvin** [comment]
Something like this maybe ?

```twig
<ul>
{% for category in post.categories %}
  {% if category.isLeaf() %}
    <li>{{ category.name }}</li>
  {% endif %}
{% endfor %}
</ul>
```

**Reply by oim37** [reply]
Yes, really great. exactly what I need, I'm delighted!

Thanks a lot  :)
